### DISCUSSION TITLE
What's the difference between "Skia+WPF head" and "Uno Islands"

### DISCUSSION THREAD

**Discussion by felixf4xu** [unoplatform/uno#18814] [Q&A]
Hi,

I'm trying to use Uno in my new WPF project. I can start entirely new with uno and without wpf, because it is a new project, but one of the requirements is to run the new project on platforms where WPF can run, i.e., .net framework 4.x, win7. The reason of considering uno is to be prepared to move to web and uno's web assembly looks good.

Maybe first question first: what will be the solution/multiple projects structure? for these 2 requirements:
1) .net framework 4.x (win 7) and above
2) potential migration to web assembly

I have not started coding but I am considering a structure like this: is it possible?
```
wpf head
web assembly head
uno_share
```
---
I'm reading the document like this:

- Using the Skia+WPF head: https://platform.uno/docs/articles/features/using-skia-wpf.html

My understand is: WPF as a shell, then uno is inside the wpf shell, with skia as uno renderer. The keyword is `WpfHost`.

- Uno Islands : https://platform.uno/docs/articles/guides/uno-islands.html

It's similar, but the key word is `XamlHost`. In this case 

So, second question: what's the difference between those 2? Which one will have the 10M size `libSkiaSharp.dll`? or both?

**Comment by DevTKSS** [comment]
That would be what I also tryed to do and asked here #18764 
I am requiring the .net Framework for a Dependency and WPF is the only option as it seems to me, beside Winforms, but in case I understood @jeromelaban correctly, the answer for us both is: No, not possible anymore with the latest uno version, which requires .net (not Framework) to be used. @felixf4xu in case you do find some way to actually do what you described, please 🙏 let me know! That would be a dream to be able to do this

**Reply by jeromelaban** [reply]
> So, second question: what's the difference between those 2? Which one will have the 10M size libSkiaSharp.dll? or both?

The Skia+WPF head is now wrapped into the `netX.0-desktop` and it's a terminology we do not use explicitly anymore. WPF is only a shell to display the app's content, in the same way X11 or AppKit are used in their respective environments.

The Uno Islands is a way to integrate an Uno content inside an actual WPF app, in case of modernization scenarios. It's similar to the [WinUI XAML Islands](https://learn.microsoft.com/en-us/windows/apps/desktop/modernize/xaml-islands/xaml-islands).

**Comment by ghost** [comment]
WpfHost with SkiaSharp will most likely require the full libSkiaSharp.dll due to its comprehensive graphics rendering.

Uno Islands with XamlHost may not need libSkiaSharp.dll in the same way, as it's about embedding Uno components into an existing UI.

If you're planning on using Uno's cross-platform capabilities for the long term and need flexibility for future web migration, starting with Uno Islands might give you a more modular and lightweight approach. If you want to fully leverage Uno's rendering in a WPF application, WpfHost with SkiaSharp would be more appropriate but might bring along the larger dependencies.
